docs/documentation/release_notes/topics/26_2_0.adoc

```diff
@@ -96,6 +96,21 @@ For more information, check the link:https://www.keycloak.org/server/management-
 Introduced the ability to dynamically select authentication flows based on conditions such as requested scopes, ACR (Authentication Context Class Reference) and others.
 This can be achieved using link:{adminguide_link}#_client_policies[Client Policies] by combining the new `AuthenticationFlowSelectorExecutor` with conditions like the new `ACRCondition`. For more details, see the link:{adminguide_link}#_client-policy-auth-flow[{adminguide_name}].
 
+= JWT Client authentication aligned with the latest OIDC specification
+
+The latest version of the link:https://openid.net/specs/openid-connect-core-1_0-36.html#rfc.section.9[OpenID Connect core specification] changed the rules for
+audience validation in JWT client assertions for the Client Authentication methods `private_key_jwt` and `client_secret_jwt`.
+
+Previously, the `aud` claim of a JWT client assertion was loosely defined as `The Audience SHOULD be the URL of the Authorization Server's Token Endpoint`, which did not exclude the usage of other URLs.
+
+The revised OIDC Core specification uses a stricter audience check: "The Audience value MUST be the OP's Issuer Identifier passed as a string, and not a single-element array.".
+
+We adapted the implementations of `JWTClientAuthenticator` and `JWTClientSecretAuthenticator` to follow the new audience check rules.
+By default, we now require the issuer URL to be the sole value of the audience.
+
+This strict audience check can be reverted to the previous more lenient check with a new 'Lenient Audience Validation'
+configuration in the respective client authentication flows. Additionally, server administrators can use the SPI configuration options `--spi-client-authenticator-client-jwt-lenient-audience-validation=true`and `--spi-client-authenticator-client-secret-jwt-lenient-audience-validation=true` to configure the old audience validation behavior.
+
```

Comment:
@thomasdarimont We are using the upgrading notes for [breaking changes](https://github.com/keycloak/keycloak/blob/main/docs/documentation/upgrading/topics/changes/changes-26_2_0.adoc#breaking-changes). So please move this part to that section. If you want you can maintain here a little note but you can just move everything there.

Reply:
Alright, I'll move the paragraph over and add a link to the upgrading guide with more details.